components/authz-service/storage/postgres/migration/sql/75_update_roles_pols.up.sql

```diff
@@ -0,0 +1,81 @@
+-- replaces "infra:*" with "infra:nodes:*" and "infra:nodeManagers:*"
+-- removes incorrect action "telemetry:*"
+-- removes system permissions
+BEGIN;
+
+UPDATE iam_roles
+    SET
+        actions = '{
+            secrets:*:get,
+            secrets:*:list,
+            infra:nodes:get,
+            infra:nodes:list,
+            infra:nodeManagers:get,
+            infra:nodeManagers:list,
+            compliance:*:get,
+            compliance:*:list,
+            event:*:get,
+            event:*:list,
+            ingest:*:get,
+            ingest:*:list,
+            iam:projects:list,
+            iam:projects:get,
+            applications:*:get,
+            applications:*:list
+        }'
+    WHERE
+        id = 'viewer';
+
+UPDATE iam_roles
+    SET
+        actions = '{
+            infra:nodes:*,
+            infra:nodeManagers:*,
+            compliance:*,
+            event:*,
+            ingest:*,
+            secrets:*,
+            iam:projects:list,
+            iam:projects:get,
+            iam:projects:assign,
+            applications:*
+        }'
+    WHERE
+        id = 'editor';
+
+UPDATE iam_roles
+    SET
+        actions = '{
+            infra:nodes:*,
+            infra:nodeManagers:*,
+            compliance:*,
+            event:*,
+            ingest:*,
+            secrets:*,
+            iam:projects:list,
+            iam:projects:get,
+            iam:projects:assign,
+            iam:policies:list,
+            iam:policies:get,
+            iam:policyMembers:*,
+            iam:teams:list,
+            iam:teams:get,
+            iam:teamUsers:*,
+            iam:users:get,
+            iam:users:list,
+            applications:*
+        }'
+    WHERE
+        id = 'project-owner';
+
+UPDATE iam_statements
```

Comment:
Even though it's not necessary with this migration, could we change the [migration code](https://github.com/chef/automate/blob/force-upgrade/components/authz-service/storage/postgres/migration/legacy/iam.go#L206) to reflect the correct migration for reference?

Reply:
I'd prefer not to *just* change the migration for the policy in the migration code and not also change the roles, and keep this migration.

If it's really clutch that try to preserve the state of the world in the `force-upgrade` branch, I'd rather modify the migration code for policies AND roles and associated tests, and the datamigrations, and not add an additional SQL migration. Or I can pull this work out and point it to master, and then we'll need to be careful to fix the things in `force-upgrade`. 

FWIW, this is why I didn't just change the migration like we'd discussed (these mostly assume that if I change the policy in the migration code, I should also change the roles (if the reason is to preserve the state of the world))

1- since this is a *post* migration migration (not just a data migration), then it's kind of weird that we're changing both-- if I'm strictly looking at the code in the future, I'm like "why do we have this migration if all the migration code already handles this?"

2- The other thing is that the datamigrations will *overwrite* the migration code-- so we'll create the roles in the v2 upgrade, then overwrite them in the datamigrations, then overwrite them _again_ in the post v2 migrations. Not a huge deal, but feels off. 

3- this change isn't really immediately related to the force upgrade-- I've put it here only bc we'd have to do a few modifications to `force-upgrade` in order to make it happy with this change. 

I agree that it's annoying that we don't have a single source of truth, but I think that this will only be a source of truth for a moment in time, and we'll need to reference the sql migrations after that. 

What do you think?

Reply:
If there's not a functional reason to update the migration code, I'm a :bigfan: of only ever touching that code again if necessary. It's a lot of high risk code we don't wanna maintain long-term.

Reply:
sounds good